### PR TITLE
Fix InducingPointKernel memory usage and speed

### PR DESCRIPTION
**Pull request by jacobrgardner**
Right now, InducingPointKernel unintentionally gets the diagonal correction by calling the kernel on the full data and then calling `diag` rather than calling the kernel with `diag=True`:
https://github.com/cornellius-gp/gpytorch/blob/4e30c948055e20c35b11e7435e8eba673a7f3401/gpytorch/kernels/inducing_point_kernel.py#L63
 This has clearly unintended ramifications for memory usage and speed. While I was at it, I also changed `_inducing_inv_root` to take advantage of the fact that `torch.cholesky` now operates in batch mode.

### CODE REVIEW

File: gpytorch/kernels/inducing_point_kernel.py [hunk 1]
```diff
@@ -60,8 +60,8 @@ def _get_covariance(self, x1, x2):
             covar = RootLazyTensor(k_ux1.matmul(self._inducing_inv_root))
 
             # Diagonal correction for predictive posterior
-            correction = (lazify(self.base_kernel(x1, x2)).diag() - covar.diag()).clamp(0, math.inf)
-            covar = PsdSumLazyTensor(DiagLazyTensor(correction), covar)
+            correction = (self.base_kernel(x1, x2, diag=True) - covar.diag()).clamp(0, math.inf)
+            covar = covar + DiagLazyTensor(correction)
```
Comment (gpleiss) [inline gpytorch/kernels/inducing_point_kernel.py]: Why not PsdSumLazyTensor?

File: gpytorch/kernels/inducing_point_kernel.py [hunk 2]
```diff
@@ -43,9 +43,9 @@ def _inducing_inv_root(self):
             return self._cached_kernel_inv_root
         else:
             inv_roots_list = []
+            chols = torch.cholesky(add_jitter(self._inducing_mat), upper=True)
```
Comment (gpleiss) [inline gpytorch/kernels/inducing_point_kernel.py]: 👏

File: gpytorch/kernels/inducing_point_kernel.py [hunk 3]
```diff
@@ -43,9 +43,9 @@ def _inducing_inv_root(self):
...
             for i in range(self._inducing_mat.size(0)):
-                jitter_mat = add_jitter(self._inducing_mat[i])
-                chol = torch.cholesky(jitter_mat, upper=True)
+                chol = chols[i]
                 eye = torch.eye(chol.size(-1), device=chol.device)
```
Comment (gpleiss) [inline gpytorch/kernels/inducing_point_kernel.py]: Minor nit: we can create this identity matrix outside the for loop

File: gpytorch/kernels/inducing_point_kernel.py (hunk 1 above)
Comment (jacobrgardner) [reply gpytorch/kernels/inducing_point_kernel.py]: This keeps things as an AddedDiagLazyTensor, but I have no strong opinions about it -- is there a good reason to prefer PsdSumLazyTensor over AddedDiagLazyTensor given that we'd have to do a root decomposition of `covar` anyways?

Comment (gpleiss) [reply gpytorch/kernels/inducing_point_kernel.py]: `covar` is already a RootLazyTensor, so we can avoid an extra root decomposition

Comment (jacobrgardner) [reply gpytorch/kernels/inducing_point_kernel.py]: Okay, I put PsdSumLazyTensor back in and simplified the for loop in inducing_inv_root further.

**Pull request merged**